doc/expressionLanguage.md

```diff
@@ -175,3 +176,52 @@ The following table shows expressions and their expected outcomes for a measure
 | '@value * 5.2'              | 31.2                  |
 | '"Pruebas " + "De Strings"' | 'Pruebas De Strings'  |
 | '@name value is @value'     | 'DevId629 value is 6' |
+
+## <a name="ngsiv2"/> NGSIv2 support
+
+As it is explained in previous sections, expressions can have two return types: String or Number, being the former one the default. Whenever an expression is executed without error, its result will be cast to the configured type. 
+
+On one hand, in NGSIv1 since all attributes' values are of type String, in the expression parser the expression type is set always to String and the transformation of the information coming from the SouthBound is done using replace instruction. Therefore, values sent to the CB will always be Strings. This can be seen in previous examples.
+
+On the other hand, NGSIv2 fully supports all the types described in the JSON
+specification (string, number, boolean, object, array and null). Therefore, the result of an expression must be cast to the appropriate type (the type used to define the attribute) in order to avoid inconsistencies between the type field for an attribute and the type of the value that is being sent.
+
+Currently, the expression parser does not support JSON Arrays and JSON document. A new issue has been created to address this aspect https://github.com/telefonicaid/iotagent-node-lib/issues/568. For the rest of types the workflow will be the following:
+
+1. Variables will be cast to Number or String depending on the expression type.
+2. The expression will be applied
+3. The output type will be cast again to the original attribute type.
+  * If attribute type is "Integer" then the value is casted to integer (JSON number)
+  * If attribute type is "Float" then the value is casted to float (JSON number)
+  * If attribute type is "Boolean" then the value is cast to boolean (JSON boolean). In order to do this conversion, only `true` or `1` are cast to true.
+  * If attribute type is "None" then the value is cast to null (JSON null)
+
+E.g.: if a device with the following provisioning information is provisioned in the IoTAgent:
+```
+{
+   "name":"status",
+   "type":"Boolean",
+   "expression": '${@status *  20}'
+}
+```
+and a measurement with the following values arrive to the IoTAgent:
+```
+status: true
+```
+
+1. The expression `*` is a multiplication, so the expression type makes `status` to be casted to Number. The cast of `true` to number is 1 (everything with a "value" is true in JavasScript).
+2. Expression is evaluated, resulting in 20
+3. 20 is cast to `20` since Expression Plugin always use String as Expression type.
+4. The attribute type is `Boolean` so the result is casted to Boolean before sending it to CB. The cast of `20` to boolean is false (only `true` or `1` are cast to true).
+
+More examples of this workflow are presented below for the different types of attributes supported in NGSIv2 and the two possible types of expressions: Integer (arithmetic operations) or Strings.
```

Comment:
It would be clarifying to include the type of the attribute in each case, eg.:

pressure (of type "Integer): 52 -> ${trim(@pressure)} -> ${trim("52")} -> $ { "52" } -> $ { "52"} -> 52
...
update (of type "Boolean"): false -> ${@update * 20} -> ${ 0 * 20 } -> $ { 0 } -> $ { "0"} -> False

Reply:
Fixed in cb737a4b8cdbf5d0ebe51fd6b1bb69410c6a765b